ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java

```diff
@@ -38,28 +37,51 @@
           + " as there are cores you want to use, as each instance is single threaded.";
 
   public static final String LISTEN_HOST = propertyName("listen.host");
-  public static final String DEFAULT_LISTEN_HOST = "0.0.0.0";
+  public static final String DEFAULT_LISTEN_HOST = "localhost";
   public static final String LISTEN_HOST_DOC =
       "The hostname to listen on";
 
   public static final String LISTEN_PORT = propertyName("listen.port");
-  public static final int DEFAULT_LISTEN_PORT = 8089;
+  public static final int DEFAULT_LISTEN_PORT = 8088;
   public static final String LISTEN_PORT_DOC =
       "The port to listen on";
 
-  public static final String KEY_PATH = propertyName("key.path");
-  public static final String KEY_PATH_DOC =
-      "Path to key file";
+  public static final String TLS_ENABLED = propertyName("tls.enabled");
+  public static final boolean DEFAULT_TLS_ENABLED = false;
+  public static final String TLS_ENABLED_DOC =
+      "Is TLS enabled?";
 
-  public static final String CERT_PATH = propertyName("cert.path");
-  public static final String CERT_PATH_DOC =
-      "Path to cert file";
+  public static final String TLS_KEY_STORE_PATH = propertyName("tls.keystore.path");
+  public static final String TLS_KEY_STORE_PATH_DOC =
+      "Path to server key store";
+
+  public static final String TLS_KEY_STORE_PASSWORD = propertyName("tls.keystore.password");
+  public static final String TLS_KEY_STORE_PASSWORD_DOC =
+      "Password for server key store";
+
+  public static final String TLS_TRUST_STORE_PATH = propertyName("tls.truststore.path");
+  public static final String TLS_TRUST_STORE_PATH_DOC =
+      "Path to client trust store";
+
+  public static final String TLS_TRUST_STORE_PASSWORD = propertyName("tls.truststore.password");
+  public static final String TLS_TRUST_STORE_PASSWORD_DOC =
+      "Password for client trust store";
+
+  public static final String TLS_CLIENT_AUTH_REQUIRED = propertyName("tls.client.auth.required");
+  public static final boolean DEFAULT_TLS_CLIENT_AUTH_REQUIRED = false;
+  public static final String TLS_CLIENT_AUTH_REQUIRED_DOC =
+      "Is client auth required?";
 
   public static final String WORKER_POOL_SIZE = propertyName("worker.pool.size");
   public static final String WORKER_POOL_DOC =
       "Max number of worker threads for executing blocking code";
   public static final int DEFAULT_WORKER_POOL_SIZE = 100;
 
+  public static final String MAX_PUSH_QUERIES = propertyName("max.push.queries");
```

Comment:
We have a similar config for limiting the number of persistent push queries: https://github.com/confluentinc/ksql/blob/f21823fa87ae3dceab08057b2cd5e2c844e5534c/ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java#L118

If this limit is for transient push queries, we should look to use consistent naming. If its for all push queries, we should try to consolidate the configs.

Reply:
As a general thing, right now, all the new api config are prefixed with "ksql.api" - they need to have some common prefix as that's what the CommonConfig class expects. Once we've migrated all the old API stuff we can consolidate things.